LPD-30357 - Refactor stylebook import/export classes

ethib137 opened the pull request:
Resent from #784

# Original description
## Motivation
Currently, the service code responsible for import/export a stylebook was living in a web module, instead of a service module, which its not the proper place to be, since web module should interacts with service/impl modules through api.

## Proposed Solution
Move code to service module.

liferay-continuous-integration commented:
The following guidelines have been set by the owner of this repository: 
- &nbsp;&nbsp;&nbsp;&nbsp;"Thank you for sending this PR to liferay-platform-experience. If you haven't already,take this opportunity to review our [pull request guidelines](https://liferay.atlassian.net/wiki/spaces/PEDS/pages/2915598365). For additional help,please reach out via the [appropriate channels](https://liferay.atlassian.net/wiki/spaces/PEDS/overview#%3Aslack%3A-Slack-Channels)."

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

ethib137 commented:
@algobob & @thiago-buarqque 

For refactors we should be handling our commits similar to how I am in this PR.

- When files are being moved or renamed, that should be handled in 1 commit. Git is smart enough to still compare these files so it makes it easier to check that the file is really just being moved or renamed.
- No logic changes should happen in rename or refactor commits.
- Renames should be as consistent as possible. This is why I added a separate commit for [Rename function name to match the return type.](https://github.com/liferay-platform-experience/liferay-portal/pull/785/commits/8b0355cc645a594def891c9dd5d2b0c9883e721f).

FYI @algobob we had agreed upon using the name [StyleBookEntryImporterResult](https://github.com/liferay-platform-experience/liferay-portal/pull/783#discussion_r1831355310) but it wasn't included in the refactor. 

Please make sure I didn't make any mistakes or break anything. If it all looks good, feel free to send it to BPM. Make sure to let them know that it is your work and I just modified the commit history.

Thanks guys.

ethib137 commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-8.liferay.com/job/test-portal-source-format/7338/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-8.liferay.com/userContent/jobs/test-portal-source-format/builds/7338/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 5 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/e554a95b98dcfd8c00c01f5b45992b6f99fcb500">e554a95b98dcfd8c00c01f5b45992b6f99fcb500</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/ethib137/liferay-portal/tree/LPD-30357-refactor-2">LPD-30357-refactor-2</a><br/>Branch GIT ID: <a href="https://github.com/ethib137/liferay-portal/commit/8b0355cc645a594def891c9dd5d2b0c9883e721f">8b0355cc645a594def891c9dd5d2b0c9883e721f</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-8.liferay.com/job/test-portal-source-format/7338/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-8.liferay.com/userContent/jobs/test-portal-source-format/builds/7338/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-8.liferay.com/job/test-portal-source-format/7338/">test-portal-source-format#7338</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-8.liferay.com/userContent/jobs/test-portal-source-format/builds/7338/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-platform-experience/liferay-portal/pull/785">liferay-platform-experience#785</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/62336281">[master] ci:test:sf - ethib137 &gt; liferay-platform-experience - PR#785 - 2024-11-07[00:27:07]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>62336281<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-8.liferay.com/job/publish-testray-report/5284/">publish-testray-report#5284</a><br/></div></div>

ethib137 commented:
[This comment](https://github.com/liferay-core-infra/liferay-portal/pull/7606#discussion_r1831150240) is also really important to always be considering with your commits.

> Please make sure you reflect this in the commit message, so in the future when people look at this part they can also know the context.

Commit messages should always be used to explain why the changes you're making are necessary and why you're doing it the way you are doing it. Thanks!

thiago-buarqque commented:
@ethib137 In the first commit you have renamed [_importResultEntries to _styleBookEntryImporterEntries](https://github.com/liferay-platform-experience/liferay-portal/pull/785/commits/2ed910e7a8d8a16212f169aa64b32ecef385fecd#diff-9685d8d9987ffdac96645ff57a8a77f9c355b3cc620346ae94dd51e1e3616e76L81-R58) to match the actual type of the List. 

In the [second commit](https://github.com/liferay-platform-experience/liferay-portal/pull/785/commits/dc14f387aee337384b47be16cf9245de17817ab1#diff-c47adb5ad7c5a32047d098027e012d034a563c33adb5701e5fdf6c4b036b4d42L50-R48) you renamed  _getStyleBookEntryZipProcessorImportResultEntryNames to _getStyleBookEntryImporterResultNames, keeping the wrong naming and then changing it in the third commit.

Why isn't the first renaming not in the second commit? And why isn't the third commit inside the second one as well?

thiago-buarqque commented:
You didn't update all usages of `StyleBookEntryImporterEntry` to `StyleBookEntryImporterResult` in `stylebookentryimporterimpl.java`

thiago-buarqque commented:
@ethib137 I do get your point, better to review, better to read, better to understand. Thanks for you insights, I will be looking carefully towards it from now on

thiago-buarqque commented on modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/display/context/ImportStyleBookDisplayContext.java:
```diff
@@ -23,48 +23,42 @@ public ImportStyleBookDisplayContext(RenderRequest renderRequest) {
-						status) {
-
+			styleBookEntryImporterResults,
+			styleBookEntryImporterEntry -> {
```
Should be `styleBookEntryImporterResult`

ethib137 commented:
> @ethib137 In the first commit you have renamed [_importResultEntries to _styleBookEntryImporterEntries](https://github.com/liferay-platform-experience/liferay-portal/pull/785/commits/2ed910e7a8d8a16212f169aa64b32ecef385fecd#diff-9685d8d9987ffdac96645ff57a8a77f9c355b3cc620346ae94dd51e1e3616e76L81-R58) to match the actual type of the List.
> 
> In the [second commit](https://github.com/liferay-platform-experience/liferay-portal/pull/785/commits/dc14f387aee337384b47be16cf9245de17817ab1#diff-c47adb5ad7c5a32047d098027e012d034a563c33adb5701e5fdf6c4b036b4d42L50-R48) you renamed _getStyleBookEntryZipProcessorImportResultEntryNames to _getStyleBookEntryImporterResultNames, keeping the wrong naming and then changing it in the third commit.
> 
> Why isn't the first renaming not in the second commit? 

@thiago-buarqque yep I rushed this to try and give you guys an idea of what I'm looking for. @algobob can you redo the commits in this general way?

> And why isn't the third commit inside the second one as well?

This is a different type of rename since it's not just changing the name because the Class changed. We are changing this name because it had the wrong name for the return type. It's best to draw attention to why we are changing it, so that it's more explicit.

So any time there's a different reason for why you are refactoring something it should be in a different commit so that the commit message can explain the reason.

algobob commented:
Stepping back from this PR for now to handle other story. I'll jump back after.
